system/lib/pthread/pthread_create.c

```diff
@@ -51,8 +53,32 @@ void __do_cleanup_pop(struct __ptcb *cb) {
   __pthread_self()->cancelbuf = cb->__next;
 }
 
-int __pthread_create(pthread_t *thread, const pthread_attr_t *attr, void *(*start_routine) (void *), void *arg) {
-  return __pthread_create_js(thread, attr, start_routine, arg);
+int __pthread_create(pthread_t *restrict res, const pthread_attr_t *restrict attrp, void *(*entry)(void *), void *restrict arg) {
+  if (!res) {
+    return EINVAL;
+  }
+
+  // Allocate thread block (pthread_t structure).
+  struct pthread *new = malloc(sizeof(struct pthread));
+  // zero-initialize thread structure.
+  memset(new, 0, sizeof(struct pthread));
+
+  // The pthread struct has a field that points to itself - this is used as a
+  // magic ID to detect whether the pthread_t structure is 'alive'.
+  new->self = new;
+  new->tid = (uintptr_t)new;
+
+  // pthread struct robust_list head should point to itself.
+  new->robust_list.head = &new->robust_list;
```

Comment:
According to `__init_tp` in musl, this should assign to `&new->robust_list.head`.
https://github.com/emscripten-core/emscripten/blob/4d864df0a57024d667e8db171aab2c3385d04c30/system/lib/libc/musl/src/env/__init_tls.c#L20
```suggestion
  new->robust_list.head = &new->robust_list.head;
```